lib/vrl/compiler/src/expression/predicate.rs

```diff
@@ -20,10 +20,26 @@ pub struct Predicate {
 impl Predicate {
     pub fn new(node: Node<Vec<Expr>>, state: (&LocalEnv, &ExternalEnv)) -> Result {
         let (span, exprs) = node.take();
-        let type_def = exprs
+        let (type_def, value) = exprs
             .last()
-            .map(|expr| expr.type_def(state))
-            .unwrap_or_else(TypeDef::null);
+            .map(|expr| (expr.type_def(state), expr.as_value()))
+            .unwrap_or_else(|| (TypeDef::null(), None));
+
+        match value {
+            Some(Value::Boolean(true)) => {
+                return Err(Error {
+                    variant: ErrorVariant::AlwaysTrue,
+                    span,
+                })
+            }
+            Some(Value::Boolean(false)) => {
+                return Err(Error {
+                    variant: ErrorVariant::AlwaysFalse,
+                    span,
+                })
+            }
+            _ => {}
+        }
```

Comment:
I just noticed that I need to do some more work here. After we introduced the concept of "warnings" in the compiler, I realized that, while this error gets turned into a warning, the rest of the predicate is no longer compiled, so you end up with an invalid program, even though the compiler succeeds in its compilation.